src/chat/actions.js

```diff
@@ -9,14 +9,27 @@ export const addChat = (chat) => async (dispatch) => {
     });
 }
 
-export const deleteChat = (chat) => async (dispatch)=> {
-    await request("DELETE", "/chats/id/" + chat._id.toString());
+export const deleteChat = (chatId) => async (dispatch, getState)=> {
+    const {chats} = getState();
+    let newCurrentChatId;
+    if(chats.length === 1) {
+        newCurrentChatId = null;
+    } else if(chats[0]._id === chatId) {
+        newCurrentChatId = chats[1]._id;
+    } else {
+        newCurrentChatId = chats[0]._id;
+    }
+    dispatch({
+        type: "SET_CURRENT_CHAT_ID",
+        chatId: newCurrentChatId,
+    });
     dispatch({
        type: "DELETE_CHAT_SUCCESS",
-       chat,
+       chatId,
     });
-    
+    await request("DELETE", "/chats/id/" + chatId.toString());
```

Comment:
Put the HTTP request before dispatching the `SET_CURRENT_CHAT_ID` and `DELETE_CHAT_SUCCESS` actions. It included `SUCCESS` to indicate that it worked correctly, but that is not known until after making the request. The UI should not indicate success if it failed.

Also, calling `.toString()` is unnecessary on the client since all ids are strings, unlike on the server when they are sometimes `ObjectId`s instead of strings.